src/Build/BackEnd/Components/RequestBuilder/TaskHost.cs

```diff
@@ -795,15 +839,18 @@ internal void MarkAsInactive()
                 VerifyActiveProxy();
                 _activeProxy = false;
 
+                ReleaseAllCores();
```

Comment:
Do you think there should be some mechanism for doing something similar when someone yields? Also wondering if it might make sense to make IBuildEngines Disposable so we could have a predefined set of things that happen whenever an IBuildEngine stops being used for any reason.

Reply:
No, yielding does not really release the cores - a build tool is running and using them. The yielding just means that msbuild can use the node for something else.